app/models/loan.rb

```diff
@@ -59,8 +59,13 @@ class Loan < ActiveRecord::Base
   belongs_to :representative, class_name: 'Person'
   belongs_to :organization_snapshot
 
-  has_many :project_steps, as: :project
-  has_many :project_logs, through: :project_steps
+  has_many :timeline_entries, as: :project
+  has_many :project_logs, through: :timeline_entries
+
+  # Do regular ruby select, to avoid issues with AR caching
+  def project_steps
+    timeline_entries.select{ |e| e.type == "ProjectStep" }
```

Comment:
@hooverlunch where do we fall on string literals? https://github.com/bbatsov/ruby-style-guide#strings I prefer option B, but most projects I have worked on prefer A.